src/lib.rs

```diff
@@ -452,17 +452,41 @@ impl Maildir {
 
     /// Copies a message from the current maildir to the targetted maildir.
     pub fn copy_to(&self, id: &str, target: &Maildir) -> std::io::Result<()> {
-        let src = self.path.join("cur").join(id);
-        let dst = target.path.join("cur").join(id);
-        fs::copy(src, dst)?;
+        let entry = self.find(id).ok_or_else(|| {
+            std::io::Error::new(std::io::ErrorKind::NotFound, "Mail entry not found")
+        })?;
+        let filename = entry
+            .path()
+            .file_name()
+            .ok_or_else(|| {
+                std::io::Error::new(
+                    std::io::ErrorKind::InvalidData,
+                    "Invalid mail entry file name",
+                )
+            })?
+            .to_string_lossy()
+            .to_string();
```

Comment:
You shouldn't need to convert the `OsStr` into a `String` here; when building the destination path the `OsStr` should be used.

Reply:
Sorry for not being as clear here, it's the `to_string_lossy` call that needs to be removed. We don't want to have a lossy conversion when there's no need for it; the original file name can be used. Ditto for the move function.